src/java.base/share/classes/java/io/SequenceInputStream.java

```diff
@@ -240,12 +240,18 @@ public void close() throws IOException {
     public long transferTo(OutputStream out) throws IOException {
         Objects.requireNonNull(out, "out");
         if (getClass() == SequenceInputStream.class) {
-            long c = 0;
+            long transferred = 0;
             while (in != null) {
-                c += in.transferTo(out);
+                if (transferred < Long.MAX_VALUE) {
+                    try {
+                        transferred = Math.addExact(transferred, in.transferTo(out));
+                    } catch (ArithmeticException ignore) {
+                        return Long.MAX_VALUE;
```

Comment:
@bplb , this looks like a bug to me: once `transferred` reaches `Long.MAX_VALUE` the transfer loop will terminate and the transfer would stop even in the case there are more streams available in the sequence.

I think it the proper code should be `transferred = Long.MAX_VALUE` instead of `return Long.MAX_VALUE`.

What do you think?

Reply:
> What do you think?

I think that you are looking at an obsolete repository:

https://github.com/openjdk/jdk/blob/c328f9589ddc3a981a2c63801bd991f8e593e69f/src/java.base/share/classes/java/io/InputStream.java#L802

Reply:
I mean SequenceInputStream, not the base class: https://github.com/openjdk/jdk/blob/c328f9589ddc3a981a2c63801bd991f8e593e69f/src/java.base/share/classes/java/io/SequenceInputStream.java#L249

Could you please double-check?

Reply:
IMHO @vlsi is right. It is incorrect that we stop the transfer in the overflow case. We should fix it as he suggested. I can do that if you like.

Reply:
Right, the base class. The suggested change was made for `InputStream` in 254d6990bcf75700f1c3aa18e0f8b73b639d9bad. It looks like it should be in `SequenceInputStream` as well. I created [JDK-8322141](https://bugs.openjdk.org/browse/JDK-8322141) to track this.

Reply:
We don't have any tests to check the behavior of the transferTo overrides when they transfer more than nine quintillion bytes so dependent on catching issues like this during code review.

Reply:
The solution proposed by @vlsi and outlined by @bplb in [JDK-8322141](https://bugs.openjdk.org/browse/JDK-8322141) now can be reviewed in https://github.com/openjdk/jdk/pull/17119. I suggest to continue the discussion *there*.